doc/arch/020-data-api.md

```diff
@@ -0,0 +1,108 @@
+- Patient lists
+    - [Patient list](https://api-sandbox.simple.org/reports/patient_lists/ch-cascabel-village?report_scope=facility)
+    - [Patient list with med history](https://api-sandbox.simple.org/reports/patient_lists/ch-cascabel-village?medication_history=true&report_scope=facility)
+    - [Overdue list](https://api-sandbox.simple.org/appointments.csv?district_slug=alder-county&facility_id=73fafb72-0a8a-43c5-b668-3fcc4bb5541c&per_page=20+per+page&search_filters%5B%5D=only_less_than_year_overdue)
+- Region reports
+    - [Whatsapp graphics](https://api-sandbox.simple.org/reports/regions/facility/ch-cascabel-village/graphics.png?quarter=4&year=2021)
+    - [Quarterly cohort report](https://api-sandbox.simple.org/reports/regions/facility/ch-cascabel-village/download.csv?period=quarter)
+    - [Monthly cohort report](https://api-sandbox.simple.org/reports/regions/facility/ch-cascabel-village/download.csv?period=month)
+    - [Monthly district report](https://api-sandbox.simple.org/reports/regions/district/ashoka-south/monthly_district_report.zip)
+    - [Monthly facility data](https://api-sandbox.simple.org/reports/regions/district/ashoka-south/monthly_district_data_report.csv)
+- Drug Stock reports
+    - [Drug stock](https://api-sandbox.simple.org/my_facilities/drug_stocks.csv)
+    - [Drug consumption](https://api-sandbox.simple.org/my_facilities/drug_consumption.csv)
+
+
+## Decision
+In order to be
+[sustainable](https://docs.google.com/document/d/11R8fx9v46DSRtVOZpAWU2JXqY0NxFA0PKOnzr3RdDpk/edit),
+we need to enable the ops team to create the custom CSVs and answer
+one-off requests independently.
+
+A Data API, which exposes _all_ the reporting data that Simple uses to
+build the dashboards might be a good way to create this self-serving
+mechanism. The idea is to provide all the data in the hands of the
+users, so that they may customize it as they see fit.
+
+Since we aren't seeing direct benefits of the API without the
+downloads interface, this will live only inside the dashboard to begin
+with.
+
+We will create the following APIs:
+- /data/v1/patients.csv:
+    - all the data in `reporting_patient_states`
+    - some history of observations (BP, BS, PD, App)
+    - HTN risk level (to be rolled into patient_states)
+    - overdue days (to be rolled into patient_states)
+- /data/v1/regions.csv:
+    - Has all the data in `reporting_facility_states` /
+      `reporting_quarterly_facility_states`
+- /data/v1/drug-stocks.csv:
+    - Has all the data in `drug_stocks`, denormalized with `protocol_drugs`
+
+They will all be filterable using the parameters:
+- `region_id`: uuid of a state, district, block, or facility region. Should this be a list?
+- `period_type`: month/quarter
+- `month_date/period`: range in period_type
+
+All the APIs will go through the same authenticationi layer, which
+decides access to:
+- APIs
+- Regions
+
+### Documentation
+We will have a translation layer that renames the columns to keys in
```

Comment:
Could we just `humanize` the column names, since we can put the DB annotations into, say, the second row of the CSV for each column?